test/python/test_extensions_simulator.py

```diff
@@ -14,26 +14,16 @@
 import qiskit.extensions.simulator
 from qiskit.tools.qi.qi import state_fidelity
 from qiskit.wrapper import execute
-from qiskit.backends.local import QasmSimulatorCpp
-from .common import QiskitTestCase
+from .common import QiskitTestCase, requires_cpp_simulator
 
 
-# Cpp backend required
-try:
-    cpp_backend = QasmSimulatorCpp()
-except FileNotFoundError:
-    _skip_class = True
-else:
-    _skip_class = False
-
-
-@unittest.skipIf(_skip_class, 'C++ simulators unavailable')
```

Comment:
If you go with the new implementation, we can simply replace this line with the new decorator.